manifests/init.pp

```diff
@@ -51,73 +51,53 @@
 # Copyright 2013 Thomas Linkin, unless otherwise noted.
 #
 class domain_membership (
-  $domain,
-  $username,
-  $password,
-  $secure_password = false,
-  $machine_ou      = undef,
-  $resetpw         = true,
-  $reboot          = true,
-  $reboot_apply    = 'finished',
-  $join_options    = '1',
-  $user_domain     = undef,
+  Stdlib::Fqdn $domain,
+  String $username,
+  Variant[String,Sensitive] $password,
+  Optional[Stdlib::Fqdn] $user_domain           = undef,
```

Comment:
In order to use this, the minimum version of stdlib required is `4.25.0`. Can you also update metadata.json in the module to reflect that?

Reply:
Good point.